lens-labels/src/Lens/Labels/Function.hs

```diff
@@ -0,0 +1,16 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+------------------------------------------------------------------------------
+-- | This module provides OverloadedLabels 'IsLabel' support to lenses
+-- generated by protolens; eg, it allows you to reference a proto field 'foo'
+-- as @#foo@ without having protolens steal perfectly good identifier names.
+module Lens.Labels.Function where
+
+import GHC.OverloadedLabels (IsLabel (..))
+import Lens.Labels (LensFn (..))
+
+instance IsLabel x (LensFn p q) => IsLabel x (p -> q) where
```

Comment:
Looks like this breaks on 8.2.1 due to a change to IsLabel:
https://travis-ci.org/google/proto-lens/jobs/308690066

You can copy the CPP logic from here: https://github.com/google/proto-lens/blob/master/lens-labels/src/Lens/Labels.hs#L83

And test whether that fixed it with `stack test --pedantic --resolver=nightly-2017-07-31`